src/config/sqs.ts

```diff
@@ -0,0 +1,105 @@
+import AWS, {AWSError} from "aws-sdk";
+import Logger from "bunyan"
+import {getLogger} from "./logger"
+import {Message, ReceiveMessageResult} from "aws-sdk/clients/sqs";
+import envVars from "./env";
+
+// All our queues must be in the same region so we can set a global parameter here
+AWS.config.update({region: envVars.SQS_WEBHOOKSQS_QUEUE_REGION || "us-west-1"});
+
+// Create SQS service client
+const sqs = new AWS.SQS({apiVersion: "2012-11-05"});
+const logger = getLogger("config.sqs")
+
+// Replace with your account id and the queue name you setup
+
+export class SqsQueue {
+	queueUrl: string;
+	queueName: string;
+
+	constructor(queueName: string, queueUrl: string) {
+		this.queueUrl = queueUrl;
+		this.queueName = queueName;
+	}
+
+	sendMessage(payload: any, log?: Logger) {
+		log = log || logger
+		const params = {
+			MessageBody: JSON.stringify(payload),
+			QueueUrl: this.queueUrl
+		};
+		sqs.sendMessage(params, (err, data) => {
+			if (err) {
+				log.error(err, "Error sending SQS message");
+			} else {
+				log.info(`Successfully added message to sqs queue messageId: ${data.MessageId}`);
+			}
+		});
+	}
+
+	deleteMessage(message: Message) {
+		const deleteParams = {
+			QueueUrl: this.queueUrl,
+			ReceiptHandle: message.ReceiptHandle
+		};
+		sqs.deleteMessage(deleteParams, (err, data) => {
+			if (err) {
+				logger.error({err, data}, "Error deleting message from the queue");
+			} else {
+				logger.debug({data}, "Successfully deleted message from queue");
+			}
+		});
+	}
+
+	executeMessage(message: Message, messageHandler) {
+		const messagePayload = JSON.parse(message.Body);
+		logger.info({messagePayload}, "Sqs message received");
+
+		try {
+			messageHandler({data: messagePayload})
+			this.deleteMessage(message)
+		} catch (err) {
+			logger.error(err, "error executing sqs message")
+		}
+	}
+
+	async handleSqsResponse(err: AWSError, data: ReceiveMessageResult, messageHandler) {
+		if (err) {
+			logger.error({err}, `Error receiving message from SQS queue, queueName ${this.queueName}`);
+		} else {
+			if (!data.Messages) {
+				logger.debug("Nothing to process");
+				return;
+			}
+
+			await Promise.all(data.Messages.map(message => {
+				return new Promise<void>((resolve) => {
+					this.executeMessage(message, messageHandler)
+					resolve()
+				})
+			})
+			)
+		}
+	}
+
+	listen(messageHandler) {
+		// Setup the receiveMessage parameters
+		const params = {
+			QueueUrl: this.queueUrl,
+			MaxNumberOfMessages: 1,
+			VisibilityTimeout: 0,
+			WaitTimeSeconds: 5
+		};
+		sqs.receiveMessage(params, async (err, data) => {
+			try {
+				await this.handleSqsResponse(err, data, messageHandler)
+			} finally {
+				this.listen(messageHandler)
```

Comment:
I need some guidance here `receiveMessage` reads messages once and I've added this call to call listen function again to query the next batch.

Reply:
What are we hoping the outcome will be if there's any error?

Reply:
This will absolutely spam SQS constantly until messages are there.  I'd recommend we look into using long polling and using something like [sqs-consumer](https://github.com/bbc/sqs-consumer) to handle our requests.  It'll make it much simpler using a well known pattern.

Reply:
I was thinking about using this library but after our experience with Bull I'd prefer to reduce amount of third party libs we use.

We already have long polling, currently set to 5 seconds, so we wouldn't spam too much until there are messages on the queue (and it is only on debug level)